src/main/java/wolox/training/validators/YearValidator.java

```diff
@@ -0,0 +1,17 @@
+package wolox.training.validators;
+
+import com.google.common.base.Preconditions;
+import wolox.training.exceptions.SingletonInstanceException;
+
+public final class YearValidator {
+
+    // This class is a singleton
+    private YearValidator() throws SingletonInstanceException {
```

Comment:
Technically the class isn't a singleton, because you don't have a method that instantiates the class and returns the single instance. You're not working with an instance, but rather with a static method of the class. I think adding a private constructor would suffice with no need for throwing an exception, but it's a personal preference 😅

Reply:
This is just to prevent a `YearValidator.class.newInstance()` (which at least used to work even if the empty constructor was private) call, but I can remove them, it doesn't hurt either.

Reply:
I believe in reality people use enums as singletons for that reasons, like:

```java
public enum Validator {
 INSTANCE;

  // instance methods
}
```

Which allows to never ever be possible to have another instance.